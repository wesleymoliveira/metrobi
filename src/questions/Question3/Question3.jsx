import React from "react";
import { Hero, Sidebar, MainContent, ExtraContent, RelatedImages, RelatedPosts, BaseLayout } from "../../components";

import * as S from "./Question3.styled";


export const Question3 = () => {
	return (
			<BaseLayout>
				<S.ContentTopRow data-testid='question3'>
					<S.LeftColumn>
						<Hero />
						<Sidebar />
					</S.LeftColumn>

					<S.RightColumn>
						<MainContent />
						<ExtraContent />
					</S.RightColumn>

				</S.ContentTopRow>

				<S.ContentBottomRow>
						<RelatedImages />
						<RelatedPosts />
				</S.ContentBottomRow>
			</BaseLayout>
	);
};