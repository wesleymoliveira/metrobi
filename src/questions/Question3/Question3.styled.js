import styled, { css } from "styled-components";

export const ContentTopRow = styled.div`
  ${({theme}) => css`
    display: flex;
    gap: ${theme.spacings.gap};
		height: 100%;
		text-decoration: none;
  `}
`;

export const ContentBottomRow = styled.div`
  ${({ theme }) => css`
    display: flex;
    gap: ${theme.spacings.gap};
  `}
`;

export const LeftColumn = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 40%;
		height: 100%;
    flex-direction: column;
		gap: ${theme.spacings.gap};
  `}
`;

export const RightColumn = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 60%;
    flex-direction: column;
    gap: ${theme.spacings.gap};
  `}
`;
