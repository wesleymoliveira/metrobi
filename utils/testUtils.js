import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
