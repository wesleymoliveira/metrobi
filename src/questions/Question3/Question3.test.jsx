import { screen } from '@testing-library/react';

import { Question3 } from './Question3';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<Question3 />', () => {
	it('should render the component', () => {
		render(<Question3 />);
		
		expect(screen.getByTestId(/question3/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<Question3 />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
