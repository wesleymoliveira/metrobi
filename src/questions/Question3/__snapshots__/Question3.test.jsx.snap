// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question3 /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100vh;
  gap: 1.0rem;
}

.c3 {
  -webkit-flex: 8;
  -ms-flex: 8;
  flex: 8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.0rem;
}

.c2 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c11 {
  color: #FAFAFA;
  font-size: 1.6rem;
}

.c1 {
  background-color: #54d5eb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  background-color: #d7c9e3;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c9 {
  background-color: #f5c632;
  -webkit-flex: 5;
  -ms-flex: 5;
  flex: 5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  justify-items: center;
  text-align: center;
}

.c7 {
  background-color: #9FC363;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c12 {
  background-color: #898989;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c14 {
  background-color: #2BB673;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c15 {
  background-color: #F1CCDE;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  text-align: center;
  overflow: hidden;
}

.c16 {
  background-color: #FFA500;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  -webkit-flex: 5;
  -ms-flex: 5;
  flex: 5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.0rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c13 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.0rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.0rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.0rem;
}

<div
  className="c0"
  data-testid="base-layout"
>
  <header
    className="c1"
    data-testid="header"
  >
    <h1
      className="c2"
      color="white"
      data-testid="typography"
      size="xxlarge"
    >
      Header
    </h1>
  </header>
  <main
    className="c3"
  >
    <div
      className="c4"
      data-testid="question3"
    >
      <div
        className="c5"
      >
        <section
          className="c6"
          data-testid="hero"
        >
          <h1
            className="c2"
            color="white"
            data-testid="typography"
            size="xxlarge"
          >
            Hero
          </h1>
        </section>
        <aside
          className="c7"
          data-testid="sidebar"
        >
          <h1
            className="c2"
            color="white"
            data-testid="typography"
            size="xxlarge"
          >
            Sidebar
          </h1>
        </aside>
      </div>
      <div
        className="c8"
      >
        <section
          className="c9"
          data-testid="main-content"
        >
          <div
            className="c10"
          >
            <h1
              className="c2"
              color="white"
              data-testid="typography"
              size="xxlarge"
            >
              Main Content
            </h1>
            <span
              className="c11"
              color="white"
              data-testid="typography"
              size="medium"
            >
              ** If thing do not look right, make sure your browser 
              <br />
               is in "Experimental Mode".
            </span>
          </div>
        </section>
        <section
          className="c12"
          data-testid="extra-content"
        >
          <h1
            className="c2"
            color="white"
            data-testid="typography"
            size="xxlarge"
          >
            Extra Content
          </h1>
        </section>
      </div>
    </div>
    <div
      className="c13"
    >
      <div
        className="c14"
        data-testid="related-images"
      >
        <h1
          className="c2"
          color="white"
          data-testid="typography"
          size="xxlarge"
        >
          Related images
        </h1>
      </div>
      <div
        className="c15"
        data-testid="related-posts"
      >
        <h1
          className="c2"
          color="white"
          data-testid="typography"
          size="xxlarge"
        >
          Related posts
        </h1>
      </div>
    </div>
  </main>
  <footer
    className="c16"
    data-testid="footer"
  >
    <h1
      className="c2"
      color="white"
      data-testid="typography"
      size="xxlarge"
    >
      Footer
    </h1>
  </footer>
</div>
`;
